### ISSUE TITLE
records: API pagination

### ISSUE THREAD

**Issue by ChiaraBi** [zenodo/zenodo#1662]
The `all_versions` parameter should be preserved in the link generation for the pagination.

**Comment by DougBurke** [comment]
Just to note that I spent a frustrating afternoon because I hadn't realized this - that is, I expected the arguments I specified in my call to be passed to the links/next field:

```
%  curl -s 'https://zenodo.org/api/records/?q=conceptrecid:"593753"&all_versions=True&sort=mostrecent' | jq '.links'
{
  "next": "https://zenodo.org/api/records/?sort=mostrecent&q=conceptrecid%3A%22593753%22&page=2&size=10",
  "self": "https://zenodo.org/api/records/?sort=mostrecent&q=conceptrecid%3A%22593753%22&page=1&size=10"
}
```

**Comment by slint** [comment]
"Truthy" values for `all_versions` should be:

- `<no value>`, i.e. if one only passes `?all_versions` without any querystring value
- `1`
- `True`/`true`/`TRUE`/`tRuE`

Anything else should be considered as "falsy" (i.e. `?all_versions=0`, `?all_versions=foobar`, etc.)

In the case where no value is passed for the parameter (i.e. plain `?all_versions`), we should automatically add `?all_versions=true` in the generated links

**Comment by slint** [comment]
Closed via #2138. We should also address https://github.com/zenodo/developers.zenodo.org/issues/26
